src/extractor/guidance/intersection_handler.cpp

```diff
@@ -383,201 +383,224 @@ std::size_t IntersectionHandler::findObviousTurn(const EdgeID via_edge,
+            !IsContinueRoad(best_option_data))
         {
-            best_deviation = deviation;
-            best = i;
+            best_option = std::distance(begin(intersection), straightest);
+            best_option_deviation =
+                angularDeviation(intersection[best_option].angle, STRAIGHT_ANGLE);
         }
     }
 
-    // We don't consider empty names a valid continue feature. This distinguishes between missing
-    // names and actual continuing roads.
-    if (in_data.name_id == EMPTY_NAMEID)
-        best_continue = 0;
-
-    if (best == 0)
+    // No non-low priority roads? Declare no obvious turn
+    if (best_option == 0)
         return 0;
 
-    const std::pair<std::int64_t, std::int64_t> num_continue_names = [&]() {
-        std::int64_t count = 0, count_valid = 0;
-        if (in_data.name_id != EMPTY_NAMEID)
-        {
-            for (std::size_t i = 1; i < intersection.size(); ++i)
-            {
-                const auto &road = intersection[i];
-                const auto &road_data = node_based_graph.GetEdgeData(road.eid);
-
-                const auto same_name = !util::guidance::requiresNameAnnounced(
-                    in_data.name_id, road_data.name_id, name_table, street_name_suffix_table);
-
-                if (same_name)
-                {
-                    ++count;
-                    if (road.entry_allowed)
-                        ++count_valid;
-                }
-            }
-        }
-        return std::make_pair(count, count_valid);
-    }();
+    auto best_continue_it =
+        std::min_element(begin(intersection), end(intersection), RoadCompareSameName);
+    if (IsContinueRoad(node_based_graph.GetEdgeData(best_continue_it->eid)))
+    {
+        best_continue = std::distance(begin(intersection), best_continue_it);
+        best_continue_deviation =
+            angularDeviation(intersection[best_continue].angle, STRAIGHT_ANGLE);
+    }
 
-    if (0 != best_continue && best != best_continue &&
-        angularDeviation(intersection[best].angle, STRAIGHT_ANGLE) <
-            MAXIMAL_ALLOWED_NO_TURN_DEVIATION &&
+    // if the best angle is going straight but the road is turning, declare no obvious turn
+    if (0 != best_continue && best_option != best_continue &&
+        best_option_deviation < MAXIMAL_ALLOWED_NO_TURN_DEVIATION &&
         node_based_graph.GetEdgeData(intersection[best_continue].eid).road_classification ==
-            node_based_graph.GetEdgeData(intersection[best].eid).road_classification)
+            best_option_data.road_classification)
     {
-        // if the best angle is going straight but the road is turning, we don't name anything
-        // obvious
         return 0;
     }
 
+    // get a count of number of ways from that intersection that qualify to have
+    // continue instruction because they share a name with the approaching way
+    std::int64_t continue_count =
```

Comment:
This (and below) could probably be `const auto`. I've done a small search, it seems not to be written anymore later on.